tensorbay/client/tests/test_gas.py

```diff
@@ -0,0 +1,281 @@
+                    "updateTime": 1622530298,
+                    "owner": "",
+                }
+            ],
+            "offset": 0,
+            "recordSize": 1,
+            "totalCount": 1,
+        }
+        open_api_do = mocker.patch(
+            f"{gas.__name__}.Client.open_api_do",
+            return_value=mock_response(data=response_data),
+        )
+        assert self.gas_client._list_datasets("test") == response_data
+        open_api_do.assert_called_once_with("GET", "", params=params)
+
+    def test_get_auth_storage_config(self, mocker):
+        with pytest.raises(TypeError):
+            self.gas_client.get_auth_storage_config("")
+
+        mocker.patch(
+            f"{gas.__name__}.Client.open_api_do",
+            return_value=mock_response(data={"configs": [], "totalCount": 0}),
+        )
+        with pytest.raises(ResourceNotExistError):
+            self.gas_client.get_auth_storage_config("cloud_config")
+
+        response_data = {
+            "configs": [
+                {
+                    "name": "cloud_config",
+                    "type": "azure",
+                    "accountName": "tensorbay",
+                    "containerName": "tensorbay",
+                },
+            ],
+            "offset": 0,
+            "recordSize": 1,
+            "totalCount": 1,
+        }
+        mocker.patch(
+            f"{gas.__name__}.Client.open_api_do",
+            return_value=mock_response(data=response_data),
+        )
+        assert (
+            self.gas_client.get_auth_storage_config("cloud_config") == response_data["configs"][0]
+        )
+
+    def test_list_auth_storage_configs(self, mocker):
+        response_data = {
+            "configs": [
+                {
+                    "name": "cloud_config",
+                    "type": "azure",
+                    "accountName": "tensorbay",
+                    "containerName": "tensorbay",
+                },
+            ],
+            "offset": 0,
+            "recordSize": 1,
+            "totalCount": 1,
+        }
+        mocker.patch(
+            f"{gas.__name__}.Client.open_api_do",
+            return_value=mock_response(data=response_data),
+        )
+        configs = self.gas_client.list_auth_storage_configs()
+
+        assert len(configs) == 1
+
+    @pytest.mark.parametrize("is_fusion", [True, False])
+    def test_create_dataset(self, mocker, is_fusion):
+        params = {"name": "test", "type": int(is_fusion), "region": "beijing"}
+        open_api_do = mocker.patch(
+            f"{gas.__name__}.Client.open_api_do",
+            return_value=mock_response(data={"id": "12345"}),
+        )
+        dataset_client = self.gas_client.create_dataset("test", is_fusion, region="beijing")
+        dataset_type = FusionDatasetClient if is_fusion else DatasetClient
+        assert isinstance(dataset_client, dataset_type)
+        assert dataset_client._name == "test"
+        assert dataset_client._dataset_id == "12345"
```

Comment:
assert dataset_client.status.branch_name == main
assert dataset_client.status.commit_id is None